### PR TITLE
Bagel changes: openjdk, jupyter notebook

### LINKED ISSUE #2947: "Everything Bagel" Jupyter notebook container does not support S3A
## What I did

1. Start `docker-compose`
1. Access http://localhost:8888 with the baked-in password ("`lakefs`")
1. Try to run this cell:
   ```py
   import pyspark
   from pyspark.sql import SparkSession

   conf = pyspark.SparkConf()
   conf.set("fs.s3a.access.key", "AKIAIOSFODNN7EXAMPLE")
   conf.set("fs.s3a.secret.key", "mykey")
   conf.set("fs.s3a.endpoint", "http://lakefs:8000")
   conf.set("fs.s3a.path.style.access", "true")
   sc = SparkSession.builder.config(conf=conf).getOrCreate()
   df = sc.sql("select 'spark' as hello ") #this statement works
   path = "s3a://testrepo/main/Sample-Spreadsheet-10-rows.csv"
   df2 = sc.read.csv(path) #this is throwing the error message. Below is the complete error log
```

## Result

Verbose error message, followed by:

``` text
Py4JJavaError: An error occurred while calling o42.csv.
: java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2667)
	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3431)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3466)
	at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
```

## Expected result

A file-not-found message from S3A filesystem.

# What have I tried

I tried to locate a hadoop-aws JAR on the container, but found none.  Given
that I am not a PySpark expert, I am at a loss what to do next.

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
Commit messages describe the changes

Closes #2947

### CODE REVIEW

File: deployments/compose/docker-compose.yml [hunk 1]
```diff
@@ -217,3 +217,7 @@ services:
     volumes:
       - ./etc/jupyter_notebook_config.py:/home/jovyan/.jupyter/jupyter_notebook_config.py
       - ./etc/hive-site.xml:/usr/local/spark/conf/hive-site.xml
+
+networks:
+  default:
+    name: bagel
```
Comment (guy-har) [inline deployments/compose/docker-compose.yml:223]: add new line

**Review by guy-har — APPROVED** [review]
LGTM 
Thanks!

**Review by johnnyaug — APPROVED** [review]

File: deployments/compose/jupyter/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM jupyter/pyspark-notebook:notebook-6.4.6
+
+ENV HADOOP_VERSION=3.3.1
+ENV AWS_SDK_VERSION=1.11.901
+ENV SPARK_HOME=/usr/local/spark-3.2.0-bin-hadoop3.2
+
+USER root
+
+RUN apt update -y && apt install -y curl
```
Comment (nopcoder) [inline deployments/compose/jupyter/Dockerfile:9]: can use wget without the package updates

File: deployments/compose/jupyter/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+FROM jupyter/pyspark-notebook:notebook-6.4.6
+
+ENV HADOOP_VERSION=3.3.1
+ENV AWS_SDK_VERSION=1.11.901
+ENV SPARK_HOME=/usr/local/spark-3.2.0-bin-hadoop3.2
```
Comment (nopcoder) [inline deployments/compose/jupyter/Dockerfile:5]: already set in the base image

File: deployments/compose/jupyter/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+FROM jupyter/pyspark-notebook:notebook-6.4.6
+
+ENV HADOOP_VERSION=3.3.1
```
Comment (nopcoder) [inline deployments/compose/jupyter/Dockerfile:3]: base image use the specific env var - choose another name

**Review by nopcoder — CHANGES_REQUESTED** [review]
minor changes

**Review by arielshaqed — APPROVED** [review]
Thanks! How was this tested? (Manually is perfectly fine)

**Comment by Jonathan-Rosenberg** [comment]
> Thanks! How was this tested? (Manually is perfectly fine)

Manually.

**Pull request merged**